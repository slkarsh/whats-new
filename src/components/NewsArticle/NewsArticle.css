.news-article {
    background-color: white;
    border: 2px solid black;
    height: 380px;
    width: 65%;
}

img {
    height: 20%;
    width: 100%
}

h3 {

}

p {

}

a {

}