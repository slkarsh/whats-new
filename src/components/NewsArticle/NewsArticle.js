import React from 'react';
import './NewsArticle.css';
 

// NEWSARTICLE COMPONENT CODE GOES HERE
const NewsArticle = ({ headline, description, imgSrc, link}) => {
    return (
        <article className='news-article'>
            <img src={imgSrc} alt={`Image of ${headline}`} />
            <h3>{headline}</h3>
            <p>{description}</p>
            <a href={link}>Read more here</a>
        </article>
    )
}

export default NewsArticle;