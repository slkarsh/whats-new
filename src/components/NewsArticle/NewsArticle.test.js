import React from 'react'
import { shallow } from 'enzyme'
import NewsArticle from './NewsArticle'

describe('NewsArticle', () => {
    it('should match the snapshot when rendered correctly', () => {
        const wrapper = shallow(<NewsArticle 
            imgSrc='www.fakeurl.com'
            headline='Today is Thursday'
            description='tomorrow will be Friday'
            link='www.linktoarticle.com'
        />)

        expect(wrapper).toMatchSnapshot()
    })
})