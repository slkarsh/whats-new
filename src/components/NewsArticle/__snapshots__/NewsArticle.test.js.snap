// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot when rendered correctly 1`] = `
<article
  className="news-article"
>
  <img
    src="www.fakeurl.com"
  />
  <h3>
    Today is Thursday
  </h3>
  <p>
    tomorrow will be Friday
  </p>
  <a
    href="www.linktoarticle.com"
  >
    Link to article
  </a>
</article>
`;
